compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4774,6 +4802,8 @@ private ParserRuleContext getNextRuleForIdentifier() {
                 return ParserRuleContext.SEMICOLON;
             case ABSOLUTE_RESOURCE_PATH:
                 return ParserRuleContext.ABSOLUTE_RESOURCE_PATH_END;
+            case FUNC_DEF_OR_FUNC_TYPE:
+                return ParserRuleContext.FUNC_DEF_OR_TYPE_DESC_RHS;
```

Comment:
Also, we will not this. As we end the context before coming here

Reply:
Fixed.